lib/logflare/backends/adaptor/webhook_adaptor.ex

```diff
@@ -59,14 +78,27 @@ defmodule Logflare.Backends.Adaptor.WebhookAdaptor do
     @moduledoc false
     use Tesla, docs: false
 
+    defguardp is_possible_pool(value)
+              when not is_nil(value) and not is_boolean(value) and is_atom(value)
+
     def send(opts) do
+      http_opt = Keyword.get(opts, :http)
+      pool_override_opt = Keyword.get(opts, :pool_override)
+
       adaptor =
-        if Keyword.get(opts, :http) == "http1" do
+        if http_opt == "http1" do
           {Tesla.Adapter.Finch, name: Logflare.FinchDefaultHttp1, receive_timeout: 5_000}
         else
           {Tesla.Adapter.Finch, name: Logflare.FinchDefault, receive_timeout: 5_000}
         end
 
+      adaptor =
+        if is_possible_pool(pool_override_opt) do
+          {Tesla.Adapter.Finch, name: pool_override_opt, receive_timeout: 5_000}
```

Comment:
```suggestion
          {Tesla.Adapter.Finch, name: pool_name, receive_timeout: 5_000}
```

Reply:
using a cond/do would make the `adaptor` binding clearer.
is_possible_pool/1 is also unnecessary as we can use `is_module/1`

Reply:
Tell me more about `is_module/1`. I'm only familiar with the compiled/loaded variants in the Code namespace, which are not ideal candidates here.